src/idempotent.ts

```diff
@@ -57,15 +61,14 @@ export class Idempotent {
           })
         );
 
-        backoff.failAfter(10);
-
         backoff.on('backoff', handler);
 
-        backoff.on('fail', async () => {
+        handler();
+
+        delay(this.options.timeout || '1m').then(() => {
+          backoff.removeAllListeners();
           reject(new Error('Timeout'));
```

Comment:
We should have a proper class for this error. So that it's easy for other developers to handle error. And not by checking the error message.